### ISSUE TITLE
MemoryError when using flow_from_directory

### ISSUE THREAD

**Issue by talhadar** [keras-team/keras#7974]
Hello,
I'm experiencing MemoryError when using flow_from_directory.
I'm using the latest keras version (version=2.0.8) with tensorflow.
My training dataset is quite large, it contains 8 sub directories of labels each contains more than a 1M files.
when running the following code:
`         print("generating train data generator")
        train_datagen = ImageDataGenerator(rescale=1./255,featurewise_center=True, featurewise_std_normalization=True)

        print("generating train batches")
        train_datagen = train_datagen.flow_from_directory(
                dataset_train_directory,
                target_size=target_size,
                batch_size=batch_size,
                shuffle=False,
                class_mode='categorical')`

i'm receiving the following error:
  File "C:\Users\...\AppData\Local\conda\conda\envs\tf\lib\site-packages\keras\preprocessing\image.py", line 480, in flow_from_directory
    follow_links=follow_links)
  File "C:\Users\...\AppData\Local\conda\conda\envs\tf\lib\site-packages\keras\preprocessing\image.py", line 993, in __init__
    for subdir in classes)))
  File "C:\Users\...\AppData\Local\conda\conda\envs\tf\lib\multiprocessing\pool.py", line 266, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "C:\Users\...\AppData\Local\conda\conda\envs\tf\lib\multiprocessing\pool.py", line 644, in get
    raise self._value
  File "C:\Users\...\AppData\Local\conda\conda\envs\tf\lib\multiprocessing\pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\...\AppData\Local\conda\conda\envs\tf\lib\multiprocessing\pool.py", line 44, in mapstar
    return list(map(*args))
  File "C:\Users\...\AppData\Local\conda\conda\envs\tf\lib\site-packages\keras\preprocessing\image.py", line 847, in _count_valid_files_in_directory
    for root, _, files in _recursive_list(directory):
  File "C:\Users\...\AppData\Local\conda\conda\envs\tf\lib\site-packages\keras\preprocessing\image.py", line 844, in _recursive_list
    return sorted(os.walk(subpath, followlinks=follow_links), key=lambda tpl: tpl[0])
  File "C:\Users\...\AppData\Local\conda\conda\envs\tf\lib\os.py", line 389, in walk
    nondirs.append(entry.name)
MemoryError

**Comment by mariokreutzfeldt** [comment]
Hi,
I noticed that you use the same name on both sides.
**train_datagen** = **train_datagen**.flow_from_directory()

I assume you used train_datagen = ImageDataGenerator() before. 
So try another name, e.g.: 
train_generator = train_datagen.flow_from_directory()

Cheers,
Mario

**Comment by ymodak** [comment]
Is this still an issue?

**Comment by ymodak** [comment]
Closing this issue due to lack of activity. Feel free to reopen when new information is available. Thanks!
